indigo_app/views/works.py

```diff
@@ -303,31 +303,93 @@ def get_top_contributors(self):
 
 
 class WorkCommencementsView(WorkViewBase, DetailView):
+    # TODO: don't forget the migration :"D
     template_name_suffix = '_commencements'
     tab = 'commencements'
 
     def get_context_data(self, **kwargs):
         context = super(WorkCommencementsView, self).get_context_data(**kwargs)
         context['provisions'] = provisions = self.work.all_commenceable_provisions()
-        context['uncommenced_provisions'] = self.work.all_uncommenced_provisions()
         context['commencements'] = commencements = self.work.commencements.all().reverse()
         context['has_all_provisions'] = any(c.all_provisions for c in commencements)
         context['has_main_commencement'] = any(c.main for c in commencements)
-        context['everything_commenced'] = context['has_all_provisions'] or (context['provisions'] and not context['uncommenced_provisions'])
+        context['uncommenced_provisions_count'], context['total_provisions_count'] = self.add_commencement_info(provisions, commencements)
+        context['everything_commenced'] = context['has_all_provisions'] or (context['provisions'] and not context['uncommenced_provisions_count'])
+
+        provision_set = {}
+
+        def add_to_set(p):
+            provision_set[p.id] = p
+            for c in p.children:
+                add_to_set(c)
+        for p in provisions:
+            add_to_set(p)
 
-        provision_set = {p.id: p for p in provisions}
         for commencement in commencements:
             # rich description of provisions
-            commencement.provision_items = [provision_set.get(p, p) for p in commencement.provisions]
+            commencement.commenced_provisions = self.get_commenced_provisions(commencement)
             # possible options
-            commencement.possible_provisions = self.get_possible_provisions(commencement, commencements, provisions)
+            commencement.possible_provisions = self.get_possible_provisions(commencement, commencements)
 
         return context
 
-    def get_possible_provisions(self, commencement, commencements, provisions):
+    def add_info(self, p, commenced_provisions, upc, tpc):
```

Comment:
This seems really similar to `get_commenced_provisions`... can the action of setting attributes be merged?